README.md

```diff
@@ -47,6 +47,92 @@ Events can be tracked without explicit parameters:
 wt.track("pageview");
 ```
 
+# React Integration
+
+`@clutter/wt` includes a React integration that supports providing event parameters via React context, validating events, and hooks to easily track events and inspect the current context parameters.
+
+WT exports a default `WTProvider` component, and two hooks: `useTrack` and `useWT`. If
+additional customization is desired, there is also a `createProvider` utility which allows for custom typing and added
+event control. The components and hooks returned by each call to `createProvider` use a separate context and cannot be
+mixed with the default exports.
+
+```tsx
+type EventParams = WTEventParams & {
+  customParamA: string;
+  customParamB?: string;
+};
+
+// The returned provider and hooks share a single context and respect the EventParams type
+const { WTProvider, useTrack, useWT } = createProvider((params: EventParams) =>
+  wt.track("event", params)
+);
+```
+
+### Context
+
+Context providers may be nested to build event parameters contextually:
+
+```tsx
+<WTProvider params={{ pageName: "home" }}>
+  <WTProvider params={{ container: "hero" }}>
+    {/* Events tracked here will receive the params { pageName: 'home', container: 'hero' } */}
+  </WTProvider>
+  <WTProvider params={{ container: "body" }}>
+    {/* Events tracked here will receive the params { pageName: 'home', container: 'body' } */}
+  </WTProvider>
+</WTProvider>
+```
+
+### Hooks
+
+`useTrack` provides the most flexible interface to track an event in a context aware manner
+
+```tsx
+function List({ children }) {
+  // Parameters passed to `useTrack` will be merged into each call to `track`
+  const track = useTrack({ container: "list" });
+
+  return (
+    <ul>
+      <li>
+        {/* Parameters can also be passed to individual `track` calls */}
+        <button onClick={() => track({ position: 1 })}>Item 1</button>
+        <button onClick={() => track({ position: 2 })}>Item 2</button>
+      </li>
+    </ul>
+  );
+}
+```
+
+`useWT` can be used if seeing the context event parameters is desired
+
+```tsx
+function List({ children }) {
+  const { track, params } = useWT();
+
+  doSomethingWithPageName(params.pageName);
+
+  return <button onClick={() => track("Tracked!")}>Track me!</button>;
```

Comment:
shouldn't an object be passed into `track`?

Reply:
🤦 why yes, it should.